### PR TITLE
Upgrade playwright to 1.0.0

### PR DESCRIPTION
**Pull request by Georgegriff**
## Motivation/Description of the PR
- Upgrade playwright to their first Major release.
- 2 breaking changes: waitFor method gone, and waitFor attribute of waitForSelector ranamed to state
- networkidle2 removed, networkidle0 === networkidle

Applicable helpers:

- [ ] WebDriver
- [ ] Puppeteer
- [ ] Nightmare
- [ ] REST
- [ ] FileHelper
- [ ] Appium
- [ ] Protractor
- [ ] TestCafe
- [x] Playwright

Applicable plugins:

- [ ] allure
- [ ] autoDelay
- [ ] autoLogin
- [ ] customLocator
- [ ] pauseOnFail
- [ ] puppeteerCoverage
- [ ] retryFailedStep
- [ ] screenshotOnFail
- [ ] selenoid
- [ ] stepByStepReport
- [ ] wdio

## Type of change

- [x] :fire: Breaking changes
- [ ] :rocket: New functionality
- [ ] :bug: Bug fix
- [x] :clipboard: Documentation changes/updates
- [ ] :hotsprings: Hot fix
- [ ] :hammer: Markdown files fix - not related to source code
- [ ] :nail_care: Polish code

## Checklist:

- [ ] Tests have been added
- [ ] Documentation has been added (Run `npm run docs`)
- [ ] Lint checking (Run `npm run lint`)
- [ ] Local tests are passed (Run `npm test`)

### CODE REVIEW

File: docs/custom-helpers.md [hunk 1]
```diff
@@ -209,7 +209,7 @@ async clickOnEveryElement(locator) {
 }
 ```
 
-In this case `el` will be an instance of [ElementHandle](https://github.com/microsoft/playwright/blob/v0.12.1/docs/api.md#class-elementhandle) which is similar for Playwright & [Puppeteer](https://pptr.dev/#?product=Puppeteer&version=v2.1.1&show=api-class-elementhandle).
+In this case `el` will be an instance of [ElementHandle](https://github.com/microsoft/playwright/blob/v1.0.1/docs/api.md#class-elementhandle) which is similar for Playwright & [Puppeteer](https://pptr.dev/#?product=Puppeteer&version=v2.1.1&show=api-class-elementhandle).
```
Comment (DavertMik) [inline docs/custom-helpers.md:212]:
```suggestion
In this case `el` will be an instance of [ElementHandle](https://playwright.dev/#version=master&path=docs%2Fapi.md&q=class-elementhandle) which is similar for Playwright & [Puppeteer](https://pptr.dev/#?product=Puppeteer&version=master&show=api-class-elementhandle).
```
Let's stick to latest documentation for not to update this links on each minor release

File: docs/helpers/Playwright.md [hunk 2]
```diff
@@ -19,9 +19,9 @@ Uses [Playwright][1] library to run tests inside:
 
 This helper works with a browser out of the box with no additional tools required to install.
 
-Requires `playwright` package version ^0.12.1 to be installed:
+Requires `playwright` package version ^1.0.1 to be installed:
 
-    npm i playwright@^0.12.1 --save
+    npm i playwright@^1.0.1 --save
```
Comment (DavertMik) [inline docs/helpers/Playwright.md:24]:
```suggestion
    npm i playwright@^1 --save
```
I think we can be less specific than 1.0.1

File: docs/helpers/Playwright.md [hunk 3]
```diff
@@ -19,9 +19,9 @@ Uses [Playwright][1] library to run tests inside:
 
 This helper works with a browser out of the box with no additional tools required to install.
 
-Requires `playwright` package version ^0.12.1 to be installed:
+Requires `playwright` package version ^1.0.1 to be installed:
```
Comment (DavertMik) [inline docs/helpers/Playwright.md:22]:
```suggestion
Requires `playwright` package to be installed:
```

File: docs/helpers/Playwright.md [hunk 4]
```diff
@@ -1755,7 +1755,7 @@ I.waitUrlEquals('http://127.0.0.1:8000/info');
 
 [11]: https://codecept.io/helpers/FileSystem
 
-[12]: https://github.com/microsoft/playwright/blob/v0.12.1/docs/api.md#browsernewpageoptions
+[12]: https://github.com/microsoft/playwright/blob/v1.0.1/docs/api.md#browsernewpageoptions
```
Comment (DavertMik) [inline docs/helpers/Playwright.md:1758]:
```suggestion
[12]: https://github.com/microsoft/playwright/blob/master/docs/api.md#browsernewpageoptions
```

File: docs/playwright.md [hunk 5]
```diff
@@ -28,7 +28,7 @@ It's readable and simple and working using Playwright API!
 To start you need CodeceptJS with Playwright packages installed
 
 ```bash
-npm install codeceptjs playwright@^0.12.1 --save
+npm install codeceptjs playwright@^1.0.1 --save
```
Comment (DavertMik) [inline docs/playwright.md:31]:
```suggestion
npm install codeceptjs playwright@^1 --save
```

File: docs/playwright.md [hunk 6]
```diff
@@ -236,7 +236,7 @@ helpers: {
   }
 }
 ```
-To adjust browser settings you can pass [custom options](https://github.com/microsoft/playwright/blob/v0.12.1/docs/api.md#browsernewcontextoptions)
+To adjust browser settings you can pass [custom options](https://github.com/microsoft/playwright/blob/v1.0.1/docs/api.md#browsernewcontextoptions)
```
Comment (DavertMik) [inline docs/playwright.md:238]:
```suggestion
To adjust browser settings you can pass [custom options](https://github.com/microsoft/playwright/blob/master/docs/api.md#browsernewcontextoptions)
```

File: docs/playwright.md [hunk 7]
```diff
@@ -263,7 +263,7 @@ Scenario('website looks nice on iPhone', () => {
 
 ## Extending
 
-Playwright has a very [rich and flexible API](https://github.com/microsoft/playwright/blob/v0.12.1/docs/api.md). Sure, you can extend your test suites to use the methods listed there. CodeceptJS already prepares some objects for you and you can use them from your you helpers.
+Playwright has a very [rich and flexible API](https://github.com/microsoft/playwright/blob/v1.0.1/docs/api.md). Sure, you can extend your test suites to use the methods listed there. CodeceptJS already prepares some objects for you and you can use them from your you helpers.
```
Comment (DavertMik) [inline docs/playwright.md:265]:
```suggestion
Playwright has a very [rich and flexible API](https://github.com/microsoft/playwright/blob/master/docs/api.md). Sure, you can extend your test suites to use the methods listed there. CodeceptJS already prepares some objects for you and you can use them from your you helpers.
```

File: docs/playwright.md [hunk 8]
```diff
@@ -300,7 +300,7 @@ async setPermissions() {
 }
 ```
 
-> [▶ Learn more about BrowserContext](https://github.com/microsoft/playwright/blob/v0.12.1/docs/api.md#class-browsercontext)
+> [▶ Learn more about BrowserContext](https://github.com/microsoft/playwright/blob/v1.0.1/docs/api.md#class-browsercontext)
```
Comment (DavertMik) [inline docs/playwright.md:302]:
```suggestion
> [▶ Learn more about BrowserContext](https://github.com/microsoft/playwright/blob/master/docs/api.md#class-browsercontext)
```

File: lib/helper/Playwright.js [hunk 9]
```diff
@@ -50,10 +50,10 @@ const { createValueEngine, createDisabledEngine } = require('./extras/Playwright
  *
  * This helper works with a browser out of the box with no additional tools required to install.
  *
- * Requires `playwright` package version ^0.12.1 to be installed:
+ * Requires `playwright` package version ^1.0.1 to be installed:
```
Comment (DavertMik) [inline lib/helper/Playwright.js:53]:
```suggestion
 * Requires `playwright` package version ^1 to be installed:
```

File: lib/helper/Playwright.js [hunk 10]
```diff
@@ -50,10 +50,10 @@ const { createValueEngine, createDisabledEngine } = require('./extras/Playwright
...
  *
  * ```
- * npm i playwright@^0.12.1 --save
+ * npm i playwright@^1.0.1 --save
```
Comment (DavertMik) [inline lib/helper/Playwright.js:56]:
```suggestion
 * npm i playwright@^1 --save
```

File: lib/helper/Playwright.js [hunk 11]
```diff
@@ -277,7 +277,7 @@ class Playwright extends Helper {
     try {
       requireg('playwright');
     } catch (e) {
-      return ['playwright@^0.12.1'];
+      return ['playwright@^1.0.1'];
```
Comment (DavertMik) [inline lib/helper/Playwright.js:280]:
```suggestion
      return ['playwright@^1'];
```

File: lib/helper/Playwright.js [hunk 12]
```diff
@@ -939,7 +939,7 @@ class Playwright extends Helper {
    * I.openNewTab();
    * ```
    *
-   * You can pass in [page options](https://github.com/microsoft/playwright/blob/v0.12.1/docs/api.md#browsernewpageoptions) to emulate device on this page
+   * You can pass in [page options](https://github.com/microsoft/playwright/blob/v1.0.1/docs/api.md#browsernewpageoptions) to emulate device on this page
```
Comment (DavertMik) [inline lib/helper/Playwright.js:942]:
```suggestion
   * You can pass in [page options](https://github.com/microsoft/playwright/blob/master/docs/api.md#browsernewpageoptions) to emulate device on this page
```

**Comment by Georgegriff** [comment]
Seems the tests are failing to start the browser + some other issues :/ not sure why. I've run out of time to look at this today

**Comment by arjun27** [comment]
Did some locally testing on the docker image that's used in Travis (`selenium/standalone-chrome:3.141.59-oxygen`), and it seems to be missing libgbm which is required by newer Chromium. Running `sudo apt install libgbm1` in the container fixes it.

To debug, you can set `DEBUG=pw:browser*` before running the test.

**Comment by DavertMik** [comment]
Thank you very much @arjun27!

**Comment by Georgegriff** [comment]
One of the builds work now, but Travis build failing with 
```
Playwright
       "before each" hook for "should open main page of configured site":
     Error: Failed to launch browser: Error: spawn /home/travis/.cache/ms-playwright/chromium-764964/chrome-linux/chrome ENOENT
```

**Comment by Georgegriff** [comment]
I've tried to fix travis based on: https://github.com/microsoft/playwright/blob/master/docs/troubleshooting.md#running-playwright-on-travis-ci but no luck :/ some help would be greatly appreciated!

**Comment by DavertMik** [comment]
Ok, the solution here would be to switch to Github Actions for Playwright unit tests.

**Comment by DavertMik** [comment]
@Georgegriff 
let's do the following

* remove Playwright from Travis CI matrix
* create a new job inside .github/workflows/playwright and name it Unit Tests
* run command for Playwright tests inside that job

**Comment by Georgegriff** [comment]
@DavertMik 
image success! The tests identified one tweak to make:
https://github.com/Codeception/CodeceptJS/pull/2383/commits/3c2650f15de2ef97591059622705bcf00639ec6f

**Comment by Georgegriff** [comment]
Looks like cicleCI had issues installing playwright 
image

**Comment by Georgegriff** [comment]
@DavertMik seems like that install error might be transient because the other docker builds worked

**Comment by Georgegriff** [comment]
If you look at the build you'll see in other docker builds in circle ci it worked :/ err was probably in downloading a browser or something

**Comment by DavertMik** [comment]
> err was probably in downloading a browser or something

Yeah, I think so... How the unit test are different here, not sure (

**Comment by Georgegriff** [comment]
It's quite possible it would work if we re ran it

**Comment by Georgegriff** [comment]
All passed now

**Pull request merged**